_drafts/oryx/boilerplate.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Boilerplate in Oryx
+description: How to Create Maintainable and Upgradeable Applications
+last_updated: Apr 3, 2023
+template: concept-topic-template
+---
+
+**TLTR:**
+
+> Oryx boilerplate is provided at https://github.com/spryker/composable-frontend.
+
+## Introduction
+
+Boilerplate refers to the _template_ code that is used to generate application code which can be further customized. At first sight, changing boilerplate is convenient as the code is at hand, generated in your project repository. However, when upgrading to newer versions of the original code, it becomes challenging. If you customized the boilerplate code and want to reapply a new version of the boilerplate, you have to merge the customizations with the new version during every update. This is a time-consuming and error-prone process that can slow down your development process and increase the risk of bugs.
+
+## How we prevent boilerplate code
+
+In Oryx, we use the following tactics to prevent boilerplate code:
+
+1. Bootstrap the application from [npm packages](https://www.npmjs.com/org/spryker-oryx) instead of source.
+2. Expose a function to set up the [app orchestrator](./app-orchestrator.md) conveniently.
+3. Provide [presets](./presets.md) for the standard application setup, including the feature sets and UI themes.
+4. Provide the source code in a [public repository](https://github.com/spryker/oryx) to enable developers to read the source code.
+5. Provide configurable components and business logic.
+6. Allow for customizations with [dependency injection](./dependency-injection.md).
+
+Most of the details around these tactics can be found elsewhere in the documentation. In the following sections you read more details on the boilerplate files that are needed.
```

Comment:
What are the files needed for?

Reply:
These files are the boilerplate. What about:

> In the following sections you read more details on the files that are part of the boilerplate.